server/routes.js

```diff
@@ -106,21 +107,47 @@ export default app => {
 
   const isDevelopment = config.env === 'development';
 
+  /**
+   * GraphQL caching
+   */
+  app.use('/graphql', async (req, res, next) => {
+    const cacheKey = getGraphqlCacheKey(req);
+    const enabled = parseToBoolean(config.graphql.cache.enabled);
+    if (cacheKey && enabled) {
+      const fromCache = await cache.get(cacheKey);
+      if (fromCache) {
+        res.send(fromCache);
+        return;
+      }
+      req.cacheKey = cacheKey;
+    }
+    next();
+  });
+
   /**
    * GraphQL v1
    */
-  const graphqlServerV1 = graphqlHTTP({
-    customFormatErrorFn: error => {
-      logger.error(`GraphQL v1 error: ${error.message}`);
-      logger.debug(error);
-      return error;
-    },
+  const graphqlServerV1 = new ApolloServer({
     schema: graphqlSchemaV1,
-    pretty: isDevelopment,
-    graphiql: isDevelopment,
+    introspection: true,
+    playground: isDevelopment,
+    engine: {
+      apiKey: get(config, 'graphql.apolloEngineAPIKey'),
+    },
+    // Align with behavior from express-graphql
+    context: ({ req }) => {
+      return req;
+    },
+    formatResponse: (response, ctx) => {
+      const req = ctx.context;
+      if (req.cacheKey) {
+        cache.set(req.cacheKey, response, Number(config.graphql.cache.ttl));
```

Comment:
Is there no risk to set the response from a logged in user here? Query like `ExpensesPage` can contain private info (ie. payoutMethod, requiredLegalDocument, transaction.uuid, etc)

Reply:
I opt out authenticated users in the latest version.